pkg/store/instance.go

```diff
@@ -63,7 +63,11 @@ func Inspect(instName string) (*Instance, error) {
 	}
 	inst.Dir = instDir
 	inst.Arch = y.Arch
-	inst.SSHLocalPort = y.SSH.LocalPort
+	if *y.SSH.AutoPort {
+		inst.SSHLocalPort = osutil.CheckOrGetFreePort(y.SSH.LocalPort)
```

Comment:
We have to return the actual port in use, not a new port.

We should let hostagent speak REST API over an UNIX socket to provide `GET /v1/info` endpoint, which would contain port information. (See the guest agent code)